modeling/data.go

```diff
@@ -0,0 +1,108 @@
+/*
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package modeling
+
+import "errors"
+
+// Zone abstracts the conception of 'zone' in clouds
+type Zone struct {
+	// Number of nodes of this zone
+	Nodes int
+	// Number of endpoints of this zone
+	Endpoints int
+	// Name of this zone
+	Name string
+	// Number of pods / total number of pods of all zones
+	endpointsRatio float64
+	// Number of nodes / total number of nodes of all zones
+	nodesRatio float64
+}
+
+type regionInfo struct {
+	// Total number of nodes of all zones
+	totalNodes int
+	// Total number of endpoints of all zones
+	totalEndpoints int
+	// Detailed information of each zone, zone name - exact zone
+	zoneDetails map[string]Zone
+}
+
+// EndpointSliceGroup represents all the EndpointSlices under a same label, one
+// group may be made up by many EndpointSlices (when the number of endpoints
+// excceeds the capacity of one EndpointSlice). Since for now there is no need
+// to know how the group is composed, we keep them as a whole
+type EndpointSliceGroup struct {
+	// Label is a unique identifier for an EndpointSliceGroup. This often
+	// represents a topology label that the group will be consumed from.
+	Label string
+	// Contribution of endpoints in this group from different zones
+	Composition map[string]int
+	// Weights of endpoints in this group for different zones while routing
+	Weights map[string]float64
+}
+
+// Traffic represents the detailed traffic infomation of a zone
+type Traffic struct {
+	// Name of a specific zone
+	ZoneName string
+	// Traffic that the same zone received
+	IncomingTraffic float64
+	// Outgoing traffic distribution of the same zone
+	OutgoingTraffic map[string]float64
```

Comment:
Nit: These names are repetitive. Could likely just use `Incoming` and `Outgoing` to end up with `traffic.Incoming` instead of `traffic.IncomingTraffic`.

Reply:
fixed